src/cloud/provision/freedom-module.ts

```diff
@@ -0,0 +1,362 @@
+    // Setup Digital Ocean (SSH key + droplet)
+    }).then((actions: any) => {
+      //console.log(actions);
+      return this.doRequest_("GET", "droplets/" + this.state.cloud.vm.id);
+    // Get the droplet's configuration
+    }).then((resp: any) => {
+      this.sendStatus_("CLOUD_DONE_VM");
+      this.state.cloud.vm = resp.droplet;
+      this.state.network.ssh_port = 22;
+      // Retrieve public IPv4 address
+      for (var i = 0; i < resp.droplet.networks.v4.length; i++) {
+        if (resp.droplet.networks.v4[i].type === "public") {
+          this.state.network['ipv4'] = resp.droplet.networks.v4[i].ip_address;
+        }
+      }
+      // Retrieve public IPv6 address
+      for (var i = 0; i < resp.droplet.networks.v6.length; i++) {
+        if (resp.droplet.networks.v6[i].type === "public") {
+          this.state.network['ipv6'] = resp.droplet.networks.v6[i].ip_address;
+        }
+      }
+      console.log(this.state);
+      return this.state;
+    });
+  }
+
+  /**
+   * One-click destruction of a VM
+   * See freedom-module.json for return and error types
+   * @todo currently doesnt wait for destroy to complete before resolving
+   * @param {String} name of VM to create
+   * @return {Promise.<Object>}
+   */
+  public stop = (name: string) : Promise<Object> => {
+    return this.doRequest_("GET", "droplets").then((resp: any) => {
+      for (var i = 0; i < resp.droplets.length; i++) {
+        if (resp.droplets[i].name === name) {
+          return Promise.resolve({
+            droplet: resp.droplets[i]
+          });
+        }
+      }
+      return Promise.reject({
+        "errcode": "VM_DNE",
+        "message": "Droplet with name," + name + ", doesnt exist"
+      });
+    }).then((resp: { droplet: { id: string } } ) => {
+      return this.doRequest_("DELETE", "droplets/" + resp.droplet.id);
+    });
+  }
+
+  /**
+   * Generates an RSA keypair using forge
+   * @return {KeyPair} public and private SSH keys
+   */
+  private static generateKeyPair_ = () : KeyPair => {
+    "use strict";
+    var pair = forge.pki.rsa.generateKeyPair({bits: 2048, e: 0x10001});
+    var publicKey = forge.ssh.publicKeyToOpenSSH(pair.publicKey, '');
+    var privateKey = forge.ssh.privateKeyToOpenSSH(pair.privateKey, '');
+    return { public: publicKey, private: privateKey };
+  }
+
+  /**
+   * Dispatches status events
+   * events listed in STATUS_CODES
+   * @param {String} code one of STATUS_CODES 
+   */
+  private sendStatus_ = (code: string): void => {
+    this.dispatch("status", {
+      "code": code,
+      "message": STATUS_CODES[code]
+    });
+  }
+
+  /**
+   * Initiates a Digital Ocean oAuth flow
+   * @return {Promise.<Object>} oAuth response from Digital Ocean
+   *  {
+   *    access_token: "..",
+   *    expires_in: "..",
+   *    state: "..",
+   *    token_type: ".."
+   *  }
+   */
+  private doOAuth_ = () : Promise<Object> => {
+    return new Promise((resolve: Function, reject: Function) => {
```

Comment:
Elsewhere in this repo we have TypeScript infer the types of `resolve` and `reject`. This means you can rewrite this line as:
```typescript
return new Promise((F, R) => {
```